transformer_qa.py

```diff
@@ -0,0 +1,81 @@
+from enum import IntEnum
+from typing import List, Optional, Dict
+
+from transformers import pipeline
+
+from jina import Document, DocumentArray, Executor, requests
+
+
+class Device(IntEnum):
+    cpu = -1
+    cuda = 0
+
+
+class TransformerQAExecutor(Executor):
+    def __init__(
+        self,
+        device: str = 'cpu',
+        model_name: str = 'bert-large-uncased-whole-word-masking-finetuned-squad',
+        tokenizer_name: Optional[str] = None,
+        default_limit: int = 10,
+        **kwargs
+    ):
+        """
+        TransformerQAExecutor wraps a question-answering model from Hugging face.
+        Given some questions and paragraphs/contexts, it extracts the relevant answers
+        from the given paragraph/context.
+
+        :param device: the device to use (either cpu or cuda).
+        :param model_name: the name of the QA model to use
+        :param tokenizer_name: the name of the tokenizer to use. If not provided, model_name is used.
+        :param default_limit: a default value that the QA model uses to specify the number of answers to
+            extract from the paragraphs. It is also possible to pass `limit` in parameters to the `generate`
+            method to change the default.
+        """
+        super().__init__(**kwargs)
+        tokenizer_name = tokenizer_name or model_name
+        self.default_limit = default_limit
+        self.model = pipeline(
+            'question-answering',
+            model=model_name,
+            tokenizer=tokenizer_name,
+            device=Device[device],
+        )
+
+    @requests(on='/search')
+    def generate(
+        self, docs: DocumentArray, parameters: Optional[Dict] = None, **kwargs
+    ):
+        """
+        The method extracts the answers of a question from the given paragraph.
+
+        :param docs: a list of documents where each document has `text` as the paragraph,
+            and question in `doc.tags['question']`
+        :param parameters: user can change the default limit by passing `limit` in `parameters`
+        """
+        parameters = parameters or {}
+        limit = parameters.get('limit', self.default_limit)
+
+        for doc in docs:
```

Comment:
Do we want to also support traversal path configuration? Here we assume it's always going to be `'r'`

Reply:
Sure, this will be useful in situation where root Document is an article, and chunks are paragraphs. Please see this [commit](https://github.com/jina-ai/executor-text-transformerqa/pull/7/commits/efa59787519779d0a571072c65999ddb77aa41b2) for the change.